pkg/model/sdk_api.go

```diff
@@ -360,20 +364,23 @@ func NewSDKAPI(api *awssdkmodel.API, apiGroupSuffix string) *SDKAPI {
 }
 
 // Override the operation type and/or resource name, if specified in config
-func getOpTypeAndResourceName(opID string, cfg *ackgenconfig.Config) ([]OpType, string) {
+func getOpTypesAndResourcesMapping(opID string, cfg *ackgenconfig.Config) ([]OpType, []string) {
 	opType, resName := GetOpTypeAndResourceNameFromOpID(opID, cfg)
-	opTypes := []OpType{opType}
+	opConfig, exists := cfg.GetOperationConfig(opID)
 
-	if operationConfig, exists := cfg.GetOperationConfig(opID); exists {
-		if operationConfig.ResourceName != "" {
-			resName = operationConfig.ResourceName
-		}
-		for _, operationType := range operationConfig.OperationType {
-			opType = OpTypeFromString(operationType)
-			opTypes = append(opTypes, opType)
-		}
+	// The existance of the operation in the config file is not enough to
+	// override the operation type and/or resource name. The operation type
+	// and/or resource name must be specified in the config file.
+	if !exists || len(opConfig.ResourceName) == 0 || len(opConfig.OperationType) == 0 {
+		return []OpType{opType}, []string{resName}
+	}
```

Comment:
@Vandita2020 You approach is correct, we only missed this last bits of code ^. When a user provides an Operation configuration without `ResourceName` nor `OperationType` directives - this block doesn't get called. Causing the opTypes list to reset (L378)